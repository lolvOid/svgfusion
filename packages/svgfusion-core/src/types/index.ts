export interface ConversionOptions {
  name?: string;
  prefix?: string;
  suffix?: string;
  optimize?: boolean;
  typescript?: boolean;
  format?: 'esm' | 'cjs';
  splitColors?: boolean;
  splitStrokeWidths?: boolean;
  fixedStrokeWidth?: boolean;
  normalizeFillStroke?: boolean;
  accessibility?: boolean;
  removeComments?: boolean;
  removeDuplicates?: boolean;
  minifyPaths?: boolean;
  // Legacy support
  isFixedStrokeWidth?: boolean;
}

export interface ReactConversionOptions extends ConversionOptions {
  memo?: boolean;
  ref?: boolean;
  titleProp?: boolean;
  descProp?: boolean;
  // Advanced SVG handling options
  icon?: boolean;
  dimensions?: boolean;
  replaceAttrValues?: Record<string, string>;
  svgProps?: Record<string, string>;
  expandProps?: boolean | 'start' | 'end';
  // Native SVG props support
  nativeProps?: boolean;
  // Accessibility options
  ariaLabelledBy?: boolean;
  ariaHidden?: boolean;
  role?: 'img' | 'graphics-document' | 'graphics-symbol' | 'presentation';
}

export interface VueConversionOptions extends ConversionOptions {
  compositionApi?: boolean;
  scriptSetup?: boolean;
  // Advanced SVG handling options
  props?: boolean;
  dimensions?: boolean;
  replaceAttrValues?: Record<string, string>;
}

export interface ConversionResult {
  code: string;
  filename: string;
  componentName: string;
  colors?: string[];
}

export interface BatchConversionOptions extends ConversionOptions {
  inputDir: string;
  outputDir: string;
  recursive?: boolean;
  extensions?: string[];
  generateIndex?: boolean;
  indexFormat?: 'ts' | 'js';
  exportType?: 'named' | 'default';
  framework?: Framework;
}

export interface BatchConversionResult {
  results: ConversionResult[];
  errors: ConversionError[];
  summary: {
    total: number;
    successful: number;
    failed: number;
  };
}

export interface ConversionError {
  file: string;
  error: string;
  stack?: string;
}

export type Framework = 'react' | 'vue';
