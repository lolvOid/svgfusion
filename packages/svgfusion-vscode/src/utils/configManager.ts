import * as vscode from 'vscode';

export class ConfigManager {
  private config: vscode.WorkspaceConfiguration;

  constructor() {
    this.config = vscode.workspace.getConfiguration('svgfusion');

    // Listen to configuration changes
    vscode.workspace.onDidChangeConfiguration(e => {
      if (e.affectsConfiguration('svgfusion')) {
        this.config = vscode.workspace.getConfiguration('svgfusion');
      }
    });
  }

  getFramework(): 'react' | 'vue' | 'auto' {
    return this.config.get<'react' | 'vue' | 'auto'>('framework', 'auto');
  }

  useTypeScript(): boolean {
    return this.config.get<boolean>('typescript', true);
  }

  getOutputDirectory(): string {
    return this.config.get<string>('outputDirectory', './src/components/icons');
  }

  // React specific options
  getReactMemo(): boolean {
    return this.config.get<boolean>('react.memo', false);
  }

  getReactForwardRef(): boolean {
    return this.config.get<boolean>('react.forwardRef', false);
  }

  // Vue specific options
  getVueSfc(): boolean {
    return this.config.get<boolean>('vue.sfc', true);
  }

  getVueScriptSetup(): boolean {
    return this.config.get<boolean>('vue.scriptSetup', true);
  }

  // Transformation options
  getSplitColors(): boolean {
    return this.config.get<boolean>('transformation.splitColors', true);
  }

  getSplitStrokeWidths(): boolean {
    return this.config.get<boolean>('transformation.splitStrokeWidths', false);
  }

  getFixedStrokeWidth(): boolean {
    return this.config.get<boolean>('transformation.fixedStrokeWidth', false);
  }

  getOptimize(): boolean {
    return this.config.get<boolean>('transformation.optimize', true);
  }

  getTransformationOptions() {
    return {
      splitColors: this.getSplitColors(),
      splitStrokeWidths: this.getSplitStrokeWidths(),
      fixedStrokeWidth: this.getFixedStrokeWidth(),
      optimize: this.getOptimize(),
    };
  }

  // Helper method to update configuration
  async updateConfig(
    key: string,
    value: any,
    configurationTarget?: vscode.ConfigurationTarget
  ): Promise<void> {
    await this.config.update(key, value, configurationTarget);
  }
}
