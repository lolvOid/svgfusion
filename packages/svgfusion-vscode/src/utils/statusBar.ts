import * as vscode from 'vscode';

export class StatusBarManager {
  private statusBarItem: vscode.StatusBarItem;

  constructor() {
    this.statusBarItem = vscode.window.createStatusBarItem(
      vscode.StatusBarAlignment.Right,
      100
    );
    this.statusBarItem.name = 'SVGFusion';
  }

  updateForEditor(editor: vscode.TextEditor | undefined): void {
    if (!editor) {
      this.statusBarItem.hide();
      return;
    }

    const document = editor.document;

    if (!document.fileName.endsWith('.svg')) {
      this.statusBarItem.hide();
      return;
    }

    // Show SVGFusion status for SVG files
    this.statusBarItem.text = '$(file-media) SVGFusion';
    this.statusBarItem.tooltip = 'SVGFusion: Convert SVG to component';
    this.statusBarItem.command = 'svgfusion.convertSvg';
    this.statusBarItem.show();
  }

  updateWithProgress(message: string): void {
    this.statusBarItem.text = `$(loading~spin) ${message}`;
    this.statusBarItem.tooltip = message;
    this.statusBarItem.command = undefined;
    this.statusBarItem.show();
  }

  updateWithSuccess(message: string): void {
    this.statusBarItem.text = `$(check) ${message}`;
    this.statusBarItem.tooltip = message;
    this.statusBarItem.command = 'svgfusion.convertSvg';
    this.statusBarItem.show();

    // Reset to normal state after a delay
    setTimeout(() => {
      this.updateForEditor(vscode.window.activeTextEditor);
    }, 3000);
  }

  updateWithError(message: string): void {
    this.statusBarItem.text = `$(error) ${message}`;
    this.statusBarItem.tooltip = message;
    this.statusBarItem.command = 'svgfusion.convertSvg';
    this.statusBarItem.show();

    // Reset to normal state after a delay
    setTimeout(() => {
      this.updateForEditor(vscode.window.activeTextEditor);
    }, 5000);
  }

  dispose(): void {
    this.statusBarItem.dispose();
  }
}
