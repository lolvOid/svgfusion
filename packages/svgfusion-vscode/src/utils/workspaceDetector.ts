import * as vscode from 'vscode';
import * as path from 'path';

export class WorkspaceDetector {
  private frameworkCache: Map<string, 'react' | 'vue' | 'unknown'> = new Map();

  constructor() {
    // Clear cache when workspace folders change
    vscode.workspace.onDidChangeWorkspaceFolders(() => {
      this.frameworkCache.clear();
    });
  }

  async detectFramework(): Promise<'react' | 'vue' | 'unknown'> {
    const workspaceFolders = vscode.workspace.workspaceFolders;
    if (!workspaceFolders || workspaceFolders.length === 0) {
      return 'unknown';
    }

    // For multiple workspace folders, check the first one or the active one
    const activeEditor = vscode.window.activeTextEditor;
    let targetFolder = workspaceFolders[0];

    if (activeEditor) {
      const activeFolder = vscode.workspace.getWorkspaceFolder(
        activeEditor.document.uri
      );
      if (activeFolder) {
        targetFolder = activeFolder;
      }
    }

    const cacheKey = targetFolder.uri.fsPath;
    if (this.frameworkCache.has(cacheKey)) {
      return this.frameworkCache.get(cacheKey)!;
    }

    const detected = await this.detectFrameworkInFolder(targetFolder);
    this.frameworkCache.set(cacheKey, detected);

    return detected;
  }

  private async detectFrameworkInFolder(
    workspaceFolder: vscode.WorkspaceFolder
  ): Promise<'react' | 'vue' | 'unknown'> {
    try {
      // Check package.json first
      const packageJsonPath = path.join(
        workspaceFolder.uri.fsPath,
        'package.json'
      );
      const packageJsonUri = vscode.Uri.file(packageJsonPath);

      try {
        const packageJsonContent =
          await vscode.workspace.fs.readFile(packageJsonUri);
        const packageJson = JSON.parse(
          Buffer.from(packageJsonContent).toString('utf8')
        );

        const allDependencies = {
          ...packageJson.dependencies,
          ...packageJson.devDependencies,
          ...packageJson.peerDependencies,
        };

        // Check for React
        if (allDependencies.react || allDependencies['@types/react']) {
          return 'react';
        }

        // Check for Vue
        if (
          allDependencies.vue ||
          allDependencies['@vue/cli-service'] ||
          allDependencies.nuxt
        ) {
          return 'vue';
        }
      } catch {
        // package.json not found or invalid, continue with file-based detection
      }

      // Check for framework-specific files
      const files = await vscode.workspace.findFiles(
        new vscode.RelativePattern(workspaceFolder, '**/*.{vue,jsx,tsx}'),
        new vscode.RelativePattern(workspaceFolder, '**/node_modules/**'),
        10 // Limit to first 10 files for performance
      );

      let reactFiles = 0;
      let vueFiles = 0;

      for (const file of files) {
        const ext = path.extname(file.fsPath);
        if (ext === '.jsx' || ext === '.tsx') {
          reactFiles++;
        } else if (ext === '.vue') {
          vueFiles++;
        }
      }

      if (reactFiles > vueFiles) {
        return 'react';
      } else if (vueFiles > reactFiles) {
        return 'vue';
      }

      // Check for configuration files
      const configFiles = await Promise.all([
        this.fileExists(workspaceFolder, 'vite.config.js'),
        this.fileExists(workspaceFolder, 'vite.config.ts'),
        this.fileExists(workspaceFolder, 'next.config.js'),
        this.fileExists(workspaceFolder, 'next.config.mjs'),
        this.fileExists(workspaceFolder, 'vue.config.js'),
        this.fileExists(workspaceFolder, 'nuxt.config.js'),
        this.fileExists(workspaceFolder, 'nuxt.config.ts'),
      ]);

      const [viteJs, viteTs, nextJs, nextMjs, vueConfig, nuxtJs, nuxtTs] =
        configFiles;

      if (nextJs || nextMjs) {
        return 'react';
      }

      if (vueConfig || nuxtJs || nuxtTs) {
        return 'vue';
      }

      // Check content of Vite config for framework hints
      if (viteJs || viteTs) {
        const configPath = viteJs || viteTs;
        if (configPath) {
          const framework =
            await this.detectFrameworkFromViteConfig(configPath);
          if (framework !== 'unknown') {
            return framework;
          }
        }
      }
    } catch (error) {
      console.error('Error detecting framework:', error);
    }

    return 'unknown';
  }

  private async fileExists(
    workspaceFolder: vscode.WorkspaceFolder,
    fileName: string
  ): Promise<string | null> {
    try {
      const filePath = path.join(workspaceFolder.uri.fsPath, fileName);
      const uri = vscode.Uri.file(filePath);
      await vscode.workspace.fs.stat(uri);
      return filePath;
    } catch {
      return null;
    }
  }

  private async detectFrameworkFromViteConfig(
    configPath: string
  ): Promise<'react' | 'vue' | 'unknown'> {
    try {
      const uri = vscode.Uri.file(configPath);
      const content = await vscode.workspace.fs.readFile(uri);
      const configText = Buffer.from(content).toString('utf8');

      if (
        configText.includes('@vitejs/plugin-react') ||
        configText.includes('plugin-react')
      ) {
        return 'react';
      }

      if (
        configText.includes('@vitejs/plugin-vue') ||
        configText.includes('plugin-vue')
      ) {
        return 'vue';
      }
    } catch {
      // Error reading config file
    }

    return 'unknown';
  }

  refresh(): void {
    this.frameworkCache.clear();
  }

  // Get project info for display
  async getProjectInfo(): Promise<{
    framework: 'react' | 'vue' | 'unknown';
    hasTypeScript: boolean;
    packageManager: 'npm' | 'yarn' | 'pnpm' | 'unknown';
  }> {
    const framework = await this.detectFramework();
    const hasTypeScript = await this.hasTypeScript();
    const packageManager = await this.detectPackageManager();

    return {
      framework,
      hasTypeScript,
      packageManager,
    };
  }

  private async hasTypeScript(): Promise<boolean> {
    const workspaceFolders = vscode.workspace.workspaceFolders;
    if (!workspaceFolders) return false;

    try {
      const tsconfigFiles = await vscode.workspace.findFiles(
        '**/tsconfig.json',
        '**/node_modules/**',
        1
      );

      return tsconfigFiles.length > 0;
    } catch {
      return false;
    }
  }

  private async detectPackageManager(): Promise<
    'npm' | 'yarn' | 'pnpm' | 'unknown'
  > {
    const workspaceFolders = vscode.workspace.workspaceFolders;
    if (!workspaceFolders) return 'unknown';

    const workspaceFolder = workspaceFolders[0];

    // Check for lock files
    const lockFiles = [
      { file: 'pnpm-lock.yaml', manager: 'pnpm' as const },
      { file: 'yarn.lock', manager: 'yarn' as const },
      { file: 'package-lock.json', manager: 'npm' as const },
    ];

    for (const { file, manager } of lockFiles) {
      if (await this.fileExists(workspaceFolder, file)) {
        return manager;
      }
    }

    return 'unknown';
  }
}
