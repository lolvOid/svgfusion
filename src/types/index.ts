export interface ConversionOptions {
  name?: string;
  prefix?: string;
  suffix?: string;
  optimize?: boolean;
  typescript?: boolean;
  format?: 'esm' | 'cjs';
}

export interface ReactConversionOptions extends ConversionOptions {
  memo?: boolean;
  ref?: boolean;
  titleProp?: boolean;
  descProp?: boolean;
  // Advanced SVG handling options
  icon?: boolean;
  dimensions?: boolean;
  replaceAttrValues?: Record<string, string>;
  svgProps?: Record<string, string>;
  expandProps?: boolean | 'start' | 'end';
  // Native SVG props support
  nativeProps?: boolean;
  // Accessibility options
  ariaLabelledBy?: boolean;
  ariaHidden?: boolean;
  role?: 'img' | 'graphics-document' | 'graphics-symbol' | 'presentation';
}

export interface VueConversionOptions extends ConversionOptions {
  compositionApi?: boolean;
  scriptSetup?: boolean;
  // Advanced SVG handling options
  props?: boolean;
  dimensions?: boolean;
  replaceAttrValues?: Record<string, string>;
}

export interface ConversionResult {
  code: string;
  filename: string;
  componentName: string;
}

export interface BatchConversionOptions extends ConversionOptions {
  inputDir: string;
  outputDir: string;
  recursive?: boolean;
  extensions?: string[];
  generateIndex?: boolean;
  indexFormat?: 'ts' | 'js';
  exportType?: 'named' | 'default';
  framework?: Framework;
}

export interface BatchConversionResult {
  results: ConversionResult[];
  errors: ConversionError[];
  summary: {
    total: number;
    successful: number;
    failed: number;
  };
}

export interface ConversionError {
  file: string;
  error: string;
  stack?: string;
}

export type Framework = 'react' | 'vue';

export interface CliOptions {
  framework: Framework;
  input: string;
  output?: string;
  name?: string;
  prefix?: string;
  suffix?: string;
  optimize?: boolean;
  typescript?: boolean;
  format?: 'esm' | 'cjs';
  recursive?: boolean;
  verbose?: boolean;
  generateIndex?: boolean;
  indexFormat?: 'ts' | 'js';
  exportType?: 'named' | 'default';
}
