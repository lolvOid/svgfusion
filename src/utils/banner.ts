import figlet from 'figlet';
import { Colors } from './colors';

// Create ASCII art banner
export function createBanner(colors: Colors, text: 'SVGfusion'): string {
  const title = figlet.textSync(text, {
    font: 'ANSI Shadow',
    horizontalLayout: 'default',
    verticalLayout: 'default',
    width: 80,
    whitespaceBreak: true,
  });

  // Split title into lines for gradient effect
  const lines = title.split('\n');
  const totalLines = lines.length;
  const gradientTitle = lines
    .map((line, lineIndex) => {
      // Process each character for 45-degree gradient
      const chars = line.split('');
      const processedLine = chars
        .map((char, charIndex) => {
          // Calculate 45-degree diagonal position from bottom-left
          // Normalize coordinates: lineIndex (0 to totalLines-1), charIndex (0 to line.length-1)
          const y = (totalLines - 1 - lineIndex) / (totalLines - 1); // 0 = top, 1 = bottom (reversed)
          const x = charIndex / (line.length - 1 || 1); // 0 = left, 1 = right

          // For 45-degree gradient from bottom-left to top-right
          // The diagonal progress is (x + y) / 2, where bottom-left is 0 and top-right is 1
          const diagonalProgress = (x + y) / 2;

          if (char.trim() === '') {
            return char; // Keep whitespace as-is
          }

          // Create sophisticated gradient with soft accent transitions (blue-dominant progression)
          if (diagonalProgress <= 0.1) {
            // 0-10% - Dark blue (bottom-left)
            return `${colors.darkBlue}${char}${colors.reset}`;
          } else if (diagonalProgress <= 0.2) {
            // 10-20% - Light blue accent
            return `${colors.lightBlue}${char}${colors.reset}`;
          } else if (diagonalProgress <= 0.32) {
            // 20-32% - Medium blue
            return `${colors.mediumBlue}${char}${colors.reset}`;
          } else if (diagonalProgress <= 0.42) {
            // 32-42% - Blue-cyan blend
            return `${colors.blueCyan}${char}${colors.reset}`;
          } else if (diagonalProgress <= 0.52) {
            // 42-52% - Medium cyan accent
            return `${colors.mediumCyan}${char}${colors.reset}`;
          } else if (diagonalProgress <= 0.62) {
            // 52-62% - Dark cyan (center transition)
            return `${colors.darkCyan}${char}${colors.reset}`;
          } else if (diagonalProgress <= 0.72) {
            // 62-72% - Soft teal accent
            return `${colors.softTeal}${char}${colors.reset}`;
          } else if (diagonalProgress <= 0.82) {
            // 72-82% - Teal green
            return `${colors.tealGreen}${char}${colors.reset}`;
          } else if (diagonalProgress <= 0.92) {
            // 82-92% - Soft green accent
            return `${colors.softGreen}${char}${colors.reset}`;
          } else {
            // 92-100% - Medium green (top-right)
            return `${colors.mediumGreen}${char}${colors.reset}`;
          }
        })
        .join('');

      return processedLine;
    })
    .join('\n');

  return `
${gradientTitle}

  ${colors.gray}Transform SVG files into production-ready components${colors.reset}
            ${colors.blue}React${colors.reset} ${colors.gray}•${colors.reset} ${colors.green}Vue 3${colors.reset} ${colors.gray}•${colors.reset} ${colors.blue}TypeScript${colors.reset}
  
${colors.gray}─────────────────────────────────────────────────────────${colors.reset}
`;
}
