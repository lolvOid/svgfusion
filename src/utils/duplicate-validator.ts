import { basename, extname } from 'node:path';
import { formatComponentName } from './name';

/**
 * Information about a duplicate name conflict
 */
export interface DuplicateNameInfo {
  componentName: string;
  files: string[];
  count: number;
}

/**
 * Result of duplicate name validation
 */
export interface DuplicateValidationResult {
  hasDuplicates: boolean;
  duplicates: DuplicateNameInfo[];
  totalFiles: number;
  uniqueNames: number;
}

/**
 * Validate for duplicate component names in a list of SVG files
 */
export function validateDuplicateNames(
  svgFiles: string[],
  options: {
    prefix?: string;
    suffix?: string;
  } = {}
): DuplicateValidationResult {
  const { prefix = '', suffix = '' } = options;
  const nameMap = new Map<string, string[]>();

  // Generate component names for each file
  for (const filePath of svgFiles) {
    const filename = basename(filePath, extname(filePath));
    const componentName = formatComponentName(filename, prefix, suffix);

    if (!nameMap.has(componentName)) {
      nameMap.set(componentName, []);
    }
    const files = nameMap.get(componentName);
    if (files) {
      files.push(filePath);
    }
  }

  // Find duplicates
  const duplicates: DuplicateNameInfo[] = [];
  for (const [componentName, files] of nameMap.entries()) {
    if (files.length > 1) {
      duplicates.push({
        componentName,
        files,
        count: files.length,
      });
    }
  }

  return {
    hasDuplicates: duplicates.length > 0,
    duplicates,
    totalFiles: svgFiles.length,
    uniqueNames: nameMap.size,
  };
}

/**
 * Format duplicate validation errors for display
 */
export function formatDuplicateErrors(
  validationResult: DuplicateValidationResult
): string {
  if (!validationResult.hasDuplicates) {
    return '';
  }

  const { duplicates, totalFiles } = validationResult;
  let errorMessage = `Found ${duplicates.length} duplicate component name(s) in ${totalFiles} files:\n\n`;

  for (const duplicate of duplicates) {
    errorMessage += `⚠️  Component name "${duplicate.componentName}" conflicts:\n`;
    for (const file of duplicate.files) {
      errorMessage += `   - ${file}\n`;
    }
    errorMessage += '\n';
  }

  errorMessage +=
    'Please rename the conflicting SVG files or use different prefix/suffix options to avoid conflicts.';

  return errorMessage;
}

/**
 * Generate suggested file names to resolve conflicts
 */
export function generateConflictResolutions(
  validationResult: DuplicateValidationResult
): Map<string, string[]> {
  const resolutions = new Map<string, string[]>();

  for (const duplicate of validationResult.duplicates) {
    const suggestions: string[] = [];

    // Generate numbered suggestions
    for (let i = 0; i < duplicate.files.length; i++) {
      const originalFile = duplicate.files[i];
      const filename = basename(originalFile, extname(originalFile));
      const numberedName = `${filename}-${i + 1}${extname(originalFile)}`;
      suggestions.push(numberedName);
    }

    resolutions.set(duplicate.componentName, suggestions);
  }

  return resolutions;
}
