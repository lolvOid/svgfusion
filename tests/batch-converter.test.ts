import { BatchConversionOptions } from '../src/types/index';
import { validateDuplicateNames } from '../src/utils/duplicate-validator';

describe('BatchConverter Options', () => {
  test('should support splitColors and isFixedStrokeWidth options', () => {
    // Test that BatchConversionOptions interface supports the new options
    const options: BatchConversionOptions = {
      inputDir: '/mock/input',
      outputDir: '/mock/output',
      framework: 'react',
      splitColors: true,
      isFixedStrokeWidth: true,
      typescript: true,
      generateIndex: false,
    };

    // Verify the options are properly typed and accessible
    expect(options.splitColors).toBe(true);
    expect(options.isFixedStrokeWidth).toBe(true);
    expect(options.framework).toBe('react');
    expect(options.typescript).toBe(true);

    // Snapshot test for options structure
    expect(options).toMatchSnapshot();
  });

  test('should have optional splitColors and isFixedStrokeWidth properties', () => {
    // Test that the options are optional (can be undefined)
    const minimalOptions: BatchConversionOptions = {
      inputDir: '/mock/input',
      outputDir: '/mock/output',
      framework: 'vue',
    };

    expect(minimalOptions.splitColors).toBeUndefined();
    expect(minimalOptions.isFixedStrokeWidth).toBeUndefined();

    // Snapshot test for minimal options structure
    expect(minimalOptions).toMatchSnapshot();
  });

  test('should support all framework options with split colors', () => {
    const reactOptions: BatchConversionOptions = {
      inputDir: '/test/input',
      outputDir: '/test/output',
      framework: 'react',
      splitColors: true,
      isFixedStrokeWidth: true,
      typescript: true,
      generateIndex: true,
      indexFormat: 'ts',
      exportType: 'named',
      prefix: 'Icon',
      suffix: 'Component',
      optimize: true,
    };

    const vueOptions: BatchConversionOptions = {
      inputDir: '/test/input',
      outputDir: '/test/output',
      framework: 'vue',
      splitColors: true,
      isFixedStrokeWidth: false,
      typescript: false,
      generateIndex: true,
      indexFormat: 'js',
      exportType: 'default',
      recursive: true,
      extensions: ['.svg', '.SVG'],
    };

    // Snapshot test for full React options
    expect(reactOptions).toMatchSnapshot();

    // Snapshot test for full Vue options
    expect(vueOptions).toMatchSnapshot();
  });

  test('should validate duplicate names before processing', () => {
    const duplicateFiles = [
      '/path/to/icon.svg',
      '/another/path/icon.svg',
      '/third/path/icon.svg',
    ];

    const result = validateDuplicateNames(duplicateFiles);

    expect(result.hasDuplicates).toBe(true);
    expect(result.duplicates).toHaveLength(1);
    expect(result.duplicates[0].componentName).toBe('Icon');
    expect(result.duplicates[0].count).toBe(3);
    expect(result).toMatchSnapshot();
  });

  test('should handle batch conversion with prefix and suffix affecting duplicates', () => {
    const files = [
      '/path/to/home.svg',
      '/path/to/settings.svg',
      '/another/path/home.svg',
    ];

    const options: BatchConversionOptions = {
      inputDir: '/mock/input',
      outputDir: '/mock/output',
      framework: 'react',
      prefix: 'App',
      suffix: 'Icon',
    };

    const result = validateDuplicateNames(files, {
      prefix: options.prefix,
      suffix: options.suffix,
    });

    expect(result.hasDuplicates).toBe(true);
    expect(result.duplicates[0].componentName).toBe('AppHomeIcon');
    expect(result).toMatchSnapshot();
  });
});
