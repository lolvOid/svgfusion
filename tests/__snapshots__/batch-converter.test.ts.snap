// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`BatchConverter Options should handle batch conversion with prefix and suffix affecting duplicates 1`] = `
{
  "duplicates": [
    {
      "componentName": "AppHomeIcon",
      "count": 2,
      "files": [
        "/path/to/home.svg",
        "/another/path/home.svg",
      ],
    },
  ],
  "hasDuplicates": true,
  "totalFiles": 3,
  "uniqueNames": 2,
}
`;

exports[`BatchConverter Options should have optional splitColors and isFixedStrokeWidth properties 1`] = `
{
  "framework": "vue",
  "inputDir": "/mock/input",
  "outputDir": "/mock/output",
}
`;

exports[`BatchConverter Options should support all framework options with split colors 1`] = `
{
  "exportType": "named",
  "framework": "react",
  "generateIndex": true,
  "indexFormat": "ts",
  "inputDir": "/test/input",
  "isFixedStrokeWidth": true,
  "optimize": true,
  "outputDir": "/test/output",
  "prefix": "Icon",
  "splitColors": true,
  "suffix": "Component",
  "typescript": true,
}
`;

exports[`BatchConverter Options should support all framework options with split colors 2`] = `
{
  "exportType": "default",
  "extensions": [
    ".svg",
    ".SVG",
  ],
  "framework": "vue",
  "generateIndex": true,
  "indexFormat": "js",
  "inputDir": "/test/input",
  "isFixedStrokeWidth": false,
  "outputDir": "/test/output",
  "recursive": true,
  "splitColors": true,
  "typescript": false,
}
`;

exports[`BatchConverter Options should support splitColors and isFixedStrokeWidth options 1`] = `
{
  "framework": "react",
  "generateIndex": false,
  "inputDir": "/mock/input",
  "isFixedStrokeWidth": true,
  "outputDir": "/mock/output",
  "splitColors": true,
  "typescript": true,
}
`;

exports[`BatchConverter Options should validate duplicate names before processing 1`] = `
{
  "duplicates": [
    {
      "componentName": "Icon",
      "count": 3,
      "files": [
        "/path/to/icon.svg",
        "/another/path/icon.svg",
        "/third/path/icon.svg",
      ],
    },
  ],
  "hasDuplicates": true,
  "totalFiles": 3,
  "uniqueNames": 1,
}
`;
